File: Lab Code/26-07-2024 Counting Sort/Counting_Sort.py
def Counting_Sort(string) :
    Count_Array = [0 for i in range(256)]

    for i in string :
        Count_Array[ord(i)] = Count_Array[ord(i)] + 1
        
    sorted_string = ""
    
    for i in range(0 , len(Count_Array)) :
        sorted_string = sorted_string + Count_Array[i] * chr(i)
    
    return sorted_string

if __name__ == '__main__' :
    string = "abcednfkjfAKODPEFIWEJDLdokwidfjeiojfwclkpowekfWOIREOWPSLDKJFOE"
    result = Counting_Sort(string)
    print(result)

File: Summary/Sorting/Counting_Sort.cpp
#include<bits/stdc++.h>
using namespace std;

string Counting_Sort(string str) {
    vector<int> Count_Array(256, 0);

    for (char ch : str) {
        Count_Array[(int)ch]++;
    }

    string sorted_string = "";

    for (int i = 0; i < 256; i++) {
        sorted_string += string(Count_Array[i], (char)i);
    }

    return sorted_string;
}

int main() {
    string str = "abcednfkjfAKODPEFIWEJDLdokwidfjeiojfwclkpowekfWOIREOWPSLDKJFOE";
    string result = Counting_Sort(str);
    cout << result << endl;

    return 0;
}
